instrumentation/oshi/library/src/main/java/io/opentelemetry/instrumentation/oshi/SystemMetrics.java

```diff
@@ -30,118 +34,111 @@ public static void registerObservers() {
                 networkIf.updateAttributes();
                 long recv = networkIf.getBytesRecv();
                 long sent = networkIf.getBytesSent();
                 String device = networkIf.getName();
-                r.observe(
-                    recv, Labels.of(DEVICE_LABEL_KEY, device, DIRECTION_LABEL_KEY, "receive"));
-                r.observe(
-                    sent, Labels.of(DEVICE_LABEL_KEY, device, DIRECTION_LABEL_KEY, "transmit"));
+                r.observe(recv, Attributes.of(DEVICE_KEY, device, DIRECTION_KEY, "receive"));
+                r.observe(sent, Attributes.of(DEVICE_KEY, device, DIRECTION_KEY, "transmit"));
               }
-            })
-        .build();
+            });
 
     meter
-        .longSumObserverBuilder("system.network.packets")
+        .gaugeBuilder("system.network.packets")
+        .ofLongs()
         .setDescription("System network packets")
         .setUnit("packets")
-        .setUpdater(
+        .buildWithCallback(
             r -> {
               for (NetworkIF networkIf : hal.getNetworkIFs()) {
                 networkIf.updateAttributes();
                 long recv = networkIf.getPacketsRecv();
                 long sent = networkIf.getPacketsSent();
                 String device = networkIf.getName();
-                r.observe(
-                    recv, Labels.of(DEVICE_LABEL_KEY, device, DIRECTION_LABEL_KEY, "receive"));
-                r.observe(
-                    sent, Labels.of(DEVICE_LABEL_KEY, device, DIRECTION_LABEL_KEY, "transmit"));
+                r.observe(recv, Attributes.of(DEVICE_KEY, device, DIRECTION_KEY, "receive"));
+                r.observe(sent, Attributes.of(DEVICE_KEY, device, DIRECTION_KEY, "transmit"));
               }
-            })
-        .build();
+            });
 
     meter
-        .longSumObserverBuilder("system.network.errors")
+        .gaugeBuilder("system.network.errors")
+        .ofLongs()
         .setDescription("System network errors")
         .setUnit("errors")
-        .setUpdater(
+        .buildWithCallback(
             r -> {
               for (NetworkIF networkIf : hal.getNetworkIFs()) {
                 networkIf.updateAttributes();
                 long recv = networkIf.getInErrors();
                 long sent = networkIf.getOutErrors();
                 String device = networkIf.getName();
-                r.observe(
-                    recv, Labels.of(DEVICE_LABEL_KEY, device, DIRECTION_LABEL_KEY, "receive"));
-                r.observe(
-                    sent, Labels.of(DEVICE_LABEL_KEY, device, DIRECTION_LABEL_KEY, "transmit"));
+                r.observe(recv, Attributes.of(DEVICE_KEY, device, DIRECTION_KEY, "receive"));
+                r.observe(sent, Attributes.of(DEVICE_KEY, device, DIRECTION_KEY, "transmit"));
               }
-            })
-        .build();
+            });
 
     meter
-        .longSumObserverBuilder("system.disk.io")
+        .gaugeBuilder("system.disk.io")
```

Comment:
If you wanted to preserve this as a sum, you should (likely) be using `upDownCounter().ofLongs()...buildWithCallback`.

However, looking at the attributes you have here, I'm not convinced up-down-sum is correct.